File: frontend/ase-delivery-portal/src/repository/types/box.ts
export enum Country {
  GERMANY = "GERMANY",
}

export interface Address {
  street: string;
  city: string;
  addition?: string;
  houseNumber: number;
  zipCode: number;
  country: Country;
}

export interface AddressInput {
  street?: string;
  city?: string;
  addition?: string;
  houseNumber?: number;
  zipCode?: number;
  country?: Country;
}

export interface Box {
  id: string;
  name: string;
  raspberryPiId: string;
  address: Address;
  createdAt: Date;
  updatedAt: Date;
}

export interface BoxUpdate {
  id: string;
  name: string;
  raspberryPiId: string;
  address: AddressInput;
}

export interface BoxInput {
  name: string;
  raspberryPiId: string;
  address: AddressInput;
}

File: backend/messaging-service/src/main/java/com/asedelivery/messagingservice/model/Box.java
package com.asedelivery.messagingservice.model;

import lombok.Builder;
import lombok.Getter;
import lombok.NonNull;
import lombok.Setter;

import java.util.Date;

@Getter
@Setter
@Builder
public class Box
{
	private String id;

	private String raspberryPiId;

	@NonNull
	private Address address;

	@NonNull
	private String name;

	private Date createdAt;
	private Date updatedAt;
}
